### PR TITLE
ability to make celestial object without world

### PR DESCRIPTION
**Pull request by Mooy1**

### CODE REVIEW

File: src/main/java/io/github/addoncommunity/galactifun/core/explorer/GalacticHolder.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package io.github.addoncommunity.galactifun.core.explorer;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * Represents a GalacticComponent that olds other GalacticComponents
+ *
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/core/explorer/GalacticHolder.java:8]: *holds

File: src/main/java/io/github/addoncommunity/galactifun/base/milkyway/solarsystem/Saturn.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import io.github.addoncommunity.galactifun.api.universe.attributes.Atmosphere;
+import io.github.addoncommunity.galactifun.api.universe.attributes.DayCycle;
+import io.github.addoncommunity.galactifun.api.universe.attributes.Gravity;
+import io.github.addoncommunity.galactifun.api.universe.attributes.terrain.Terrain;
+import org.bukkit.Material;
+import org.bukkit.World;
+
+/**
+ * Saturn, an ex of a planet with no implemented world
+ */
```
Comment (Seggan) [inline src/main/java/io/github/addoncommunity/galactifun/base/milkyway/solarsystem/Saturn.java:13]: *example

File: src/main/java/io/github/addoncommunity/galactifun/core/GalacticCategory.java [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
+ * Category for exploring the universe
+ * 
+ * @author Mooy1
+ */
 public final class GalacticCategory extends FlexCategory {
 
-    public GalacticCategory(NamespacedKey key, ItemStack item) {
-        super(key, item);
+    public GalacticCategory() {
+        super(PluginUtils.getKey("galactic_category"), null);
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/core/GalacticCategory.java:20]: Should be Nonnull (based on annotation)

**Review by Seggan — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/addoncommunity/galactifun/core/GalacticListener.java [hunk 4]
```diff
@@ -29,7 +29,7 @@ private GalacticListener() {
     
     @EventHandler
     public void onPlanetChange(@Nonnull PlayerChangedWorldEvent e){
-        CelestialObject object = GalacticRegistry.getCelestialObject(e);
+        CelestialWorld object = GalacticRegistry.getCelestialObject(e);
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/core/GalacticListener.java:32]: CelestialWorld that you get by getCelestialObject is really confusing, should be changed

File: src/main/java/io/github/addoncommunity/galactifun/core/GalacticListener.java [hunk 5]
```diff
@@ -38,7 +38,7 @@ public void onPlanetChange(@Nonnull PlayerChangedWorldEvent e){
 
     @EventHandler
     public void onPlanetJoin(@Nonnull PlayerJoinEvent e) {
-        CelestialObject object = GalacticRegistry.getCelestialObject(e);
+        CelestialWorld object = GalacticRegistry.getCelestialObject(e);
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/core/GalacticListener.java:41]: Same as above

File: src/main/java/io/github/addoncommunity/galactifun/core/GalacticListener.java [hunk 6]
```diff
@@ -47,7 +47,7 @@ public void onPlanetJoin(@Nonnull PlayerJoinEvent e) {
 
     @EventHandler
     public void onPlanetRespawn(@Nonnull PlayerRespawnEvent e) {
-        CelestialObject object = GalacticRegistry.getCelestialObject(e);
+        CelestialWorld object = GalacticRegistry.getCelestialObject(e);
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/core/GalacticListener.java:50]: Same as above

File: src/main/java/io/github/addoncommunity/galactifun/core/GalacticRegistry.java [hunk 7]
```diff
@@ -37,8 +37,8 @@ public static Galaxy getGalaxy(@Nonnull String name) {
     }
 
     @Nullable
-    public static CelestialObject getCelestialObject(@Nonnull String worldName) {
-        return CELESTIAL_OBJECTS.get(worldName);
+    public static CelestialWorld getCelestialObject(@Nonnull String worldName) {
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/core/GalacticRegistry.java:40]: Should be getCelestialWorld

File: src/main/java/io/github/addoncommunity/galactifun/base/milkyway/solarsystem/Earth.java [hunk 8]
```diff
@@ -6,8 +6,9 @@
 import io.github.addoncommunity.galactifun.api.universe.attributes.Atmosphere;
 import io.github.addoncommunity.galactifun.api.universe.attributes.DayCycle;
 import io.github.addoncommunity.galactifun.api.universe.attributes.Gravity;
-import io.github.addoncommunity.galactifun.api.universe.attributes.Terrain;
+import io.github.addoncommunity.galactifun.api.universe.attributes.terrain.Terrain;
 import io.github.mooy1.infinitylib.PluginUtils;
+import org.apache.commons.lang.Validate;
```
Comment (GallowsDove) [inline src/main/java/io/github/addoncommunity/galactifun/base/milkyway/solarsystem/Earth.java:11]: Unused import

**Review by GallowsDove — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/addoncommunity/galactifun/api/universe/Galaxy.java [hunk 9]
```diff
@@ -24,10 +29,19 @@ public Galaxy(@Nonnull String name, @Nonnull StarSystem... systems) {
         this.systems = new ArrayList<>(Arrays.asList(systems));
         this.name = name;
         GalacticRegistry.register(this);
+        TheUniverse.UNIVERSE.add(this);
+    }
+    
+    @Nonnull
+    @Override
+    public ItemStack getDisplayItem() {
+        return null;
```
[the remark this answers is not in the document]
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/api/universe/Galaxy.java:37]: placeholder til i make it

File: src/main/java/io/github/addoncommunity/galactifun/core/GalacticCategory.java (hunk 3 above)
Comment (Mooy1) [reply src/main/java/io/github/addoncommunity/galactifun/core/GalacticCategory.java:20]: placeholder til i make

**Review by Seggan — APPROVED** [review]

**Review by GallowsDove — APPROVED** [review]

**Pull request merged**